Apps/Admin/Tests/Services/BetaFeatureServiceTests.cs

```diff
@@ -0,0 +1,346 @@
+
+            int expectedDeletedCount = 0;
+            if (existingBetaFeature != null && betaFeature == null)
+            {
+                // Right now there is only one Salesforce feature. When there are more features, this count can be different if test setup is changed.
+                expectedDeletedCount = betaFeatureAssociations.Count;
+            }
+
+            int expectedAddedCount = 0;
+            if (existingBetaFeature == null && betaFeature != null)
+            {
+                // Right now there is only one Salesforce feature. When there are more features, this count can be different if test setup is changed.
+                expectedAddedCount = userProfiles.Count;
+            }
+
+            // Act
+            await service.SetUserAccessAsync(Email1, betaFeatures);
+
+            // Assert
+            betaFeatureAccessDelegateMock.Verify(
+                v => v.DeleteRangeAsync(
+                    It.Is<IEnumerable<BetaFeatureAccess>>(x => x.Count() == expectedDeletedCount),
+                    false,
+                    It.IsAny<CancellationToken>()));
+
+            betaFeatureAccessDelegateMock.Verify(
+                v => v.AddRangeAsync(
+                    It.Is<IEnumerable<BetaFeatureAccess>>(x => x.Count() == expectedAddedCount),
+                    true,
+                    It.IsAny<CancellationToken>()));
+        }
+
+        /// <summary>
+        /// SetUserAccessAsync throws exception.
+        /// </summary>
+        /// <param name="expectedExceptionType">The exception type to be thrown.</param>
+        /// <param name="expectedErrorMessage">The associated error message for the exception.</param>
+        /// <param name="profileExists">Value to determine whether user profile exists or not.</param>
+        /// <param name="betaFeature">The beta feature to set.</param>
+        /// <returns>
+        /// A <see cref="Task"/> representing the asynchronous unit test.
+        /// </returns>
+        [InlineData(typeof(NotImplementedException), "Reverse mapping for Unknown is not implemented", true, Common.Constants.BetaFeature.Unknown)]
+        [InlineData(typeof(NotFoundException), ErrorMessages.UserProfileNotFound, false, Common.Constants.BetaFeature.Salesforce)]
+        [Theory]
+        public async Task SetUserAccessThrowsException(Type expectedExceptionType, string expectedErrorMessage, bool profileExists, Common.Constants.BetaFeature betaFeature)
+        {
+            IList<Common.Constants.BetaFeature> betaFeatures =
+            [
+                betaFeature,
+            ];
+
+            // Arrange
+            ICollection<BetaFeatureCode> betaFeatureCodes =
+            [
+                GenerateBetaFeatureCode(BetaFeature.Salesforce),
+            ];
+
+            IList<UserProfile> userProfiles = profileExists
+                ?
+                [
+                    GenerateUserProfile(Hdid1Email1, Email1, betaFeatureCodes),
+                ]
+                : [];
```

Comment:
Nit: formatting for readability.

```suggestion
            IList<UserProfile> userProfiles = profileExists
                ? [GenerateUserProfile(Hdid1Email1, Email1, betaFeatureCodes)]
                : [];
```